web/src/main/java/org/wso2/testgrid/web/sso/SSOSessionCheckFilter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.web.sso;
+
+import org.wso2.testgrid.common.exception.TestGridException;
+import org.wso2.testgrid.web.utils.ConfigurationContext;
+import org.wso2.testgrid.web.utils.Constants;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class check whether a session exists for the user and do the needfuls accordingly.
+ */
+public class SSOSessionCheckFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    /**
+     * Check each request's path and do check for SESSION if the path has to be secured.
+     * Otherwise allowed
+     */
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        String path = ((HttpServletRequest) servletRequest).getRequestURI();
+        if (path.startsWith(Constants.LOGIN_URI) ||
+                path.startsWith(Constants.STATIC_DATA_URI) ||
+                path.startsWith(Constants.BACKEND_APIS_URI)) {    //Until backend APIs authentication is provided.
```

Comment:
Can we move this conditions check to a separate method?

Reply:
And, why do you skip the BACKEND_APIS_URI? It is a must to secure the API. I think you only need to skip the LOGIN_URI and the ACS URI.

Reply:
Skipping APIs until #373 is addressed and completed.
Yes, we can also remove STATIC_DATA_URI, but then the static data need in the Login page (Ex: TestGrid Logo,  Login buttons in future) will also not be accessible without login. We will have to find those certain static data needs to Login page and explicitly skip those.

Reply:
OK, we can address the API as part of that.

IMO, the static data for login page should be served from the /login/* sub-context. It's ok for now though.